test/integration/apiserver/runtime_config_test.go

```diff
@@ -0,0 +1,170 @@
+// +build integration,!no-etcd
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/cmd/kube-apiserver/app"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/pkg/api/testapi"
+)
+
+// Starts the apiserver. It takes an index to start the apiserver on a different port each time.
+// Returns the server IP and a channel to stop it.
+func run(t *testing.T, runtimeConfig string, index int) (string, chan struct{}) {
```

Comment:
There are `framework.NewIntegrationTestMasterConfig()` and `framework.RunAMaster`. Why do we need yet another way to start a server?